src/include/ikarus/finiteElements/mechanics/enhancedAssumedStrains.hh

```diff
@@ -0,0 +1,406 @@
+        = toEigenMatrix(geometry.jacobianInverseTransposed(quadPos0));  // J^{-1}.Transpose() in Dune = J^{-1}
+    const auto detJ0 = geometry.integrationElement(quadPos0);           // determinant(J)
+
+    auto jaco = (jacobianinvT0).inverse().eval();
+    auto J11  = jaco(0, 0);
+    auto J12  = jaco(0, 1);
+    auto J13  = jaco(0, 2);
+    auto J21  = jaco(1, 0);
+    auto J22  = jaco(1, 1);
+    auto J23  = jaco(1, 2);
+    auto J31  = jaco(2, 0);
+    auto J32  = jaco(2, 1);
+    auto J33  = jaco(2, 2);
+
+    Eigen::Matrix<double, 6, 6> T0;
+    T0 << J11 * J11, J12 * J12, J13 * J13, J11 * J12, J11 * J13, J12 * J13, J21 * J21, J22 * J22, J23 * J23, J21 * J22,
+        J21 * J23, J22 * J23, J31 * J31, J32 * J32, J33 * J33, J31 * J32, J31 * J33, J32 * J33, 2.0 * J11 * J21,
+        2.0 * J12 * J22, 2.0 * J13 * J23, J11 * J22 + J21 * J12, J11 * J23 + J21 * J13, J12 * J23 + J22 * J13,
+        2.0 * J11 * J31, 2.0 * J12 * J32, 2.0 * J13 * J33, J11 * J32 + J31 * J12, J11 * J33 + J31 * J13,
+        J12 * J33 + J32 * J13, 2.0 * J31 * J21, 2.0 * J32 * J22, 2.0 * J33 * J23, J31 * J22 + J21 * J32,
+        J31 * J23 + J21 * J33, J32 * J23 + J22 * J33;
+
+    return T0.inverse() * detJ0;
+  }
+
+  template <typename Geometry>
+  struct EASQ1E4 {
+    static constexpr int strainSize         = 3;
+    static constexpr int enhancedStrainSize = 4;
+
+    EASQ1E4() = default;
+    explicit EASQ1E4(const Geometry& geometry)
+        : geometry{std::make_unique<Geometry>(geometry)}, T0InverseTransformed{calcTransformationMatrix2D(geometry)} {}
+
+    auto calcM(const Dune::FieldVector<double, 2>& quadPos) const {
+      Eigen::Matrix<double, strainSize, enhancedStrainSize> M;
+      M.setZero(strainSize, enhancedStrainSize);
+      const double xi   = quadPos[0];
+      const double eta  = quadPos[1];
+      M(0, 0)           = xi - 0.5;
+      M(1, 1)           = eta - 0.5;
+      M(2, 2)           = xi - 0.5;
+      M(2, 3)           = eta - 0.5;
+      const double detJ = geometry->integrationElement(quadPos);
+      M                 = T0InverseTransformed / detJ * M;
+      return M;
+    }
+
+    std::unique_ptr<Geometry> geometry;
+    Eigen::Matrix3d T0InverseTransformed;
+  };
+
+  template <typename Geometry>
+  struct EASQ1E5 {
+    static constexpr int strainSize         = 3;
+    static constexpr int enhancedStrainSize = 5;
+
+    EASQ1E5() = default;
+    explicit EASQ1E5(const Geometry& geometry)
+        : geometry{std::make_unique<Geometry>(geometry)}, T0InverseTransformed{calcTransformationMatrix2D(geometry)} {}
+
+    auto calcM(const Dune::FieldVector<double, 2>& quadPos) const {
+      Eigen::Matrix<double, strainSize, enhancedStrainSize> M;
+      M.setZero();
+      const double xi   = quadPos[0];
+      const double eta  = quadPos[1];
+      M(0, 0)           = xi - 0.5;
+      M(1, 1)           = eta - 0.5;
+      M(2, 2)           = xi - 0.5;
+      M(2, 3)           = eta - 0.5;
+      M(2, 4)           = xi * eta - 0.25;
```

Comment:
this should be further investigated. In Maple i have a file where i derived. There should be:

(2*xi-1)*(2*eta-1)

and similar below.

Tarun can you change this and rerun the cooks membrane convergence plot if the EASE4 and E5 differ?